Plugins/Events/Events/ResourceEvents.cpp

```diff
@@ -0,0 +1,161 @@
+                {
+                    processEvents = true;
+                }
+            }
+
+            auto ProcessEvent = [tasks](const std::string &eventType, const std::string &file) -> void
+            {
+                tasks->QueueOnMainThread([eventType, file]()
+                {
+                    if (Core::g_CoreShuttingDown)
+                        return;
+
+                    LOG_DEBUG("EventType: %s, File: %s", eventType, file);
+
+                    uint16_t resType = Globals::ExoResMan()->GetResTypeFromFile(file);
+
+                    if (resType != 65535)
+                    {
+                        CResRef resRef;
+                        Globals::ExoResMan()->GetResRefFromFile(resRef, file);
+
+                        Events::PushEventData("RESREF", resRef.GetResRefStr());
+                        Events::PushEventData("TYPE", std::to_string(resType));
+                        Events::SignalEvent("NWNX_ON_RESOURCE_" + eventType, Utils::GetModule()->m_idSelf);
+                    }
+                });
+            };
+
+            while (processEvents)
+            {
+                int pollNum = poll(fds, 2, -1);
+
+                if (pollNum > 0)
+                {
+                    if (fds[1].revents & POLLIN)
+                    {
+                        size_t length = read(fd, buffer, EVENT_BUF_LEN);
+
+                        if (length > 0)
+                        {
+                            for (size_t i = 0; i < length;)
+                            {
+                                auto *event = reinterpret_cast<inotify_event *>(&buffer[i]);
+
+                                if (event->len)
+                                {
+                                    if (event->mask & IN_MODIFY)
+                                    {
+                                        ProcessEvent("MODIFIED", event->name);
+                                    }
+                                    else
+                                    if (event->mask & IN_CREATE)
+                                    {
+                                        ProcessEvent("ADDED", event->name);
+                                    }
+                                    else
+                                    if (event->mask & IN_DELETE)
+                                    {
+                                        ProcessEvent("REMOVED", event->name);
+                                    }
+                                }
+
+                                i += (EVENT_SIZE + event->len);
+                            }
+                        }
+                    }
+                }
+            }
+
+            if (wd != -1)
+                inotify_rm_watch(fd, wd);
+            if (fd != -1)
+                close(fd);
+        });
+    });
+}
+
+ResourceEvents::~ResourceEvents()
+{
+    processEvents = false;
+
+    if (m_pollThread)
+    {
+        int ret = write(selfPipeFds[1], "a", 1);
```

Comment:
You can just do `(void)write(...);`

Reply:
I tried that at first but my compiler still complained. Maybe it's just extra fussy.